src/gc/proxy.d

```diff
@@ -318,6 +321,20 @@ extern (C)
         return proxy.gc_inFinalizer();
     }
 
+    void gc_lock() nothrow @nogc
+    {
+        if( proxy is null )
+            return _gc.lock();
+        return proxy.gc_lock();
+    }
+
+    void gc_unlock() nothrow @nogc
+    {
+        if( proxy is null )
+            return _gc.unlock();
+        return proxy.gc_unlock();
+    }
```

Comment:
I never understood the proxy thing of the GC, I completely got rid of it in CDGC :P